### PR TITLE
VCST-2136: add virto pages support

### PR DESCRIPTION
**Pull request by basilkot**
### Jira-link:

https://virtocommerce.atlassian.net/browse/VCST-2136
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.Contentful_3.804.0-pr-12-811c.zip

### CODE REVIEW

File: .deployment/module/argoDeploy.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "artifactKey": "VirtoCommerce.Pages",
```
Comment (ksavosteev) [inline .deployment/module/argoDeploy.json]: VirtoCommerce.Contentful

File: .deployment/module/cloudDeploy.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+    "artifactKey": "VirtoCommerce.Pages",
```
Comment (ksavosteev) [inline .deployment/module/cloudDeploy.json]: VirtoCommerce.Contentful

File: .deployment/samples/argoDeploy.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+    "artifactKey": "VirtoCommerce.PagesSample",
```
Comment (ksavosteev) [inline .deployment/samples/argoDeploy.json]: don't need this file

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Common settings
+
+.github/* @mvktsk @Vectorfield4
+.gitignore @mvktsk @Vectorfield4
+.dockerignore @mvktsk @Vectorfield4
+
+# Main Code and Tests
```
Comment (ksavosteev) [inline .github/CODEOWNERS]: old copypasted values, update to yourself of remove

File: src/VirtoCommerce.Contentful.Data/Extensions/ContentfulExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public static PageOperation ToPageOperation(this string value)
+    {
+        return value switch
+        {
+            "ContentManagement.Entry.archive" => PageOperation.Archive,
+            // "ContentManagement.Entry.save" => PageOperation.Unknown,
+            // "ContentManagement.Entry.create" => PageOperation.Unknown,
+            // "ContentManagement.Entry.auto_save" => PageOperation.Unknown,
+            // "ContentManagement.Entry.unarchive" => PageOperation.Unpublish,
```
Comment (ksavosteev) [inline src/VirtoCommerce.Contentful.Data/Extensions/ContentfulExtensions.cs]: remove or uncommit

**Review by ksavosteev — CHANGES_REQUESTED** [review]

**Review by ksavosteev — APPROVED** [review]

File: src/VirtoCommerce.Contentful.Core/VirtoCommerce.Contentful.Core.csproj [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>disable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="contentful.csharp" Version="8.3.0" />
+    <PackageReference Include="VirtoCommerce.Pages.Core" Version="3.804.0" />
+    <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.853.0" />
+    <PackageReference Include="VirtoCommerce.StoreModule.Core" Version="3.809.0" />
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Core/VirtoCommerce.Contentful.Core.csproj:12]: Add this dependencies to the module.manifest

File: src/VirtoCommerce.Contentful.Web/VirtoCommerce.Contentful.Web.csproj [hunk 7]
```diff
@@ -12,16 +12,19 @@
...
-    <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.800.0" />
-    <PackageReference Include="VirtoCommerce.ContentModule.Core" Version="3.800.0" />
-    <PackageReference Include="VirtoCommerce.CoreModule.Core" Version="3.800.0" />
-    <PackageReference Include="VirtoCommerce.StoreModule.Core" Version="3.800.0" />
-    <PackageReference Include="YamlDotNet" Version="16.1.3" />
+    <PackageReference Include="VirtoCommerce.CatalogModule.Core" Version="3.828.0" />
+    <PackageReference Include="VirtoCommerce.Platform.Core" Version="3.866.0" />
+    <PackageReference Include="VirtoCommerce.ContentModule.Core" Version="3.824.0" />
+    <PackageReference Include="VirtoCommerce.CoreModule.Core" Version="3.812.0" />
+    <PackageReference Include="VirtoCommerce.StoreModule.Core" Version="3.810.0" />
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Web/VirtoCommerce.Contentful.Web.csproj]: Add this dependencies to the module.manifest

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Contentful.Web/VirtoCommerce.Contentful.Web.csproj [hunk 8]
```diff
@@ -12,16 +12,19 @@
     <None Remove="Services\**" />
   </ItemGroup>
   <ItemGroup>
-    <Folder Include="Controllers\" />
+    <Folder Include="Extensions\" />
     <Folder Include="Properties\" />
   </ItemGroup>
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Web/VirtoCommerce.Contentful.Web.csproj]: Remove this ItemGroup

File: src/VirtoCommerce.Contentful.Web/Handlers/ContentfulHandlers.cs [hunk 9]
```diff
@@ -0,0 +1,5 @@
+namespace VirtoCommerce.Contentful.Web.Handlers;
+
+public class ContentfulHandlers
+{
+}
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Web/Handlers/ContentfulHandlers.cs]: Remove this file

File: src/VirtoCommerce.Contentful.Web/Controllers/Api/PagesContentfulController.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            pageChangedEvent.Operation = pageOperation;
+            pageChangedEvent.Page = pageDocument;
+
+            await eventPublisher.Publish(pageChangedEvent);
+        }
+
+        return Ok();
+    }
+
+}
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Web/Controllers/Api/PagesContentfulController.cs]: Remove empty line

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.Contentful.Web/module.manifest [hunk 11]
```diff
@@ -4,11 +4,13 @@
...
 
-  <platformVersion>3.800.0</platformVersion>
+  <platformVersion>3.853.0</platformVersion>
   <dependencies>
-    <dependency id="VirtoCommerce.Catalog" version="3.800.0" />
-    <dependency id="VirtoCommerce.Content" version="3.800.0" />
-    <dependency id="VirtoCommerce.Core" version="3.800.0" />
+    <dependency id="VirtoCommerce.Catalog" version="3.828.0" />
+    <dependency id="VirtoCommerce.Content" version="3.824.0" />
+    <dependency id="VirtoCommerce.Core" version="3.853.0" />
```
Comment (artem-dudarev) [inline src/VirtoCommerce.Contentful.Web/module.manifest]: 3.812.0

**Review by artem-dudarev — CHANGES_REQUESTED** [review]

**Review by artem-dudarev — APPROVED** [review]

**Pull request merged**